File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/models/Burguer.java
package Ficheros.BurguerPig.models;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

public class Burguer implements Serializable {
    private final String name;
    private final List<Ingredient> ingredients;
    private final UUID id;
    private double priceCalculate;

    public Burguer(String name, List<Ingredient> ingredients) {
        this.name = name;
        this.ingredients = ingredients;
        this.id = UUID.randomUUID();
        this.priceCalculate = ingredients.stream().mapToDouble(Ingredient::getPrice).sum() * 1.5;
    }

    public Burguer(String name, List<Ingredient> ingredients, UUID id, double priceCalculate) {
        this.name = name;
        this.ingredients = ingredients;
        this.id = id;
        this.priceCalculate = priceCalculate;
    }

    public String getName() {
        return name;
    }

    public List<Ingredient> getIngredients() {
        return ingredients;
    }

    public UUID getUuid() {
        return id;
    }

    public double getPrice() {
        return priceCalculate;
    }

    @Override
    public String toString() {
        return "Burguer{" +
                "name='" + name + '\'' +
                ", ingredients=" + ingredients +
                ", id=" + id +
                ", priceCalculate=" + priceCalculate +
                '}';
    }

    public UUID getUUID() {
        return id;
    }
}

File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Kotlin/src/main/kotlin/Ficheros/BurguerPig/models/Burguer.kt
package Ficheros.BurguerPig.models

import java.io.Serializable
import java.util.*

class Burguer(val name: String, val ingredients: List<Ingredient>) : Serializable {
    private var id: UUID
    fun getUUID(): UUID {
        return id
    }

    private var priceCalculate: Double
    fun getPrice(): Double {
        return priceCalculate
    }

    constructor(name: String, ingredients: List<Ingredient>, id: UUID, priceCalculate: Double) : this(
        name,
        ingredients
    ) {
        this.id = id
        this.priceCalculate = priceCalculate
    }

    init {
        id = UUID.randomUUID()
        priceCalculate = ingredients.sumOf { it.price } * 1.5
    }

    override fun toString(): String {
        return "Burguer(name='$name', ingredients=$ingredients, id=$id, priceCalculate=$priceCalculate)"
    }

}